stream-rpc/src/subscriptions.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    connection::Client,
+    counters,
+    errors::JsonRpcError,
+    json_rpc::{
+        Method, RequestIdentifier, SubscribeResult, SubscribeToEventsParams,
+        SubscribeToTransactionsParams,
+    },
+};
+use diem_json_rpc::data::{get_events, get_transactions};
+use diem_logger::{debug, warn};
+use rand::Rng;
+use std::{borrow::Borrow, sync::Arc};
+use storage_interface::DbReader;
+use tokio::task::JoinHandle;
+
+static TIME_TO_SLEEP: u64 = 1000;
+
+pub async fn sleep_wiggled(time_ms: Option<u64>, wiggle_ms: Option<u64>) {
+    let base_time = time_ms.unwrap_or(TIME_TO_SLEEP);
+    let time_to_sleep = base_time
+        + rand::thread_rng().gen_range(0..wiggle_ms.unwrap_or((base_time as f64 * 0.1) as u64));
+    tokio::time::sleep(tokio::time::Duration::from_millis(time_to_sleep)).await;
+}
+
+pub fn subscribe_to_transactions(
+    db: &Arc<dyn DbReader>,
```

Comment:
originally I had the reference because before the task is spawned (that requires the db move) I wanted to leave it up to specific handler method to clone for two cases:
1. this isn't a subscription, and so we can query directly on the db and send the client the results without having to clone/move the db, as there is no async function here
2. The handler method could do validation on supplied params, and potentially reject them, and so no clone would be required

Should I still clone before passing in or leave as is?

Reply:
got it, feel free to leave it as is for now then